File: Mobzy-API/src/main/java/com/offz/spigot/mobzy/Pathfinders/Flying/PathfinderGoalFlyDamageTarget.java
package com.offz.spigot.mobzy.Pathfinders.Flying;

import com.offz.spigot.mobzy.Mobs.Types.FlyingMob;
import net.minecraft.server.v1_13_R2.PathfinderGoal;
import org.bukkit.entity.LivingEntity;

public class PathfinderGoalFlyDamageTarget extends PathfinderGoal {
    private FlyingMob mob;
    private long hitAt;
    private long cooldown = 200;

    public PathfinderGoalFlyDamageTarget(FlyingMob flyingMob) {
        mob = flyingMob;
    }

    public boolean a() {
        return true;
    }

    public void e() {
        if (mob.getGoalTarget() != null && hitAt + cooldown < System.currentTimeMillis()) {
            LivingEntity target = (LivingEntity) mob.getGoalTarget().getBukkitEntity();
            Double attackDamage = mob.getStaticBuilder().getAttackDamage();
            //if within range, harm
            if (attackDamage != null && mob.distanceToEntity(target) < mob.width / 2 + 1) {
                hitAt = System.currentTimeMillis();
                target.damage(attackDamage.floatValue(), mob.getBukkitEntity());
            }
        }
    }
}

File: Mobzy-API/src/main/java/com/offz/spigot/mobzy/pathfinders/flying/PathfinderGoalFlyDamageTarget.kt
package com.offz.spigot.mobzy.pathfinders.flying

import com.offz.spigot.mobzy.mobs.types.FlyingMob
import com.offz.spigot.mobzy.pathfinders.MobzyPathfinderGoal

class PathfinderGoalFlyDamageTarget(mob: FlyingMob) : MobzyPathfinderGoal(mob) {
    private var cooldown: Int = 0
    override fun shouldExecute(): Boolean {
        return target != null && --cooldown < 0
    }

    override fun shouldKeepExecuting(): Boolean = target != null

    override fun execute() {
        cooldown = 20
        val target = target ?: return
        val attackDamage: Double = mob.staticTemplate.attackDamage ?: return
        //if within range, harm
        if (mob.canReach(target)) target.damage(attackDamage, entity)
    }
}